src/app/Command.cpp

```diff
@@ -131,11 +132,8 @@ void Command::Shutdown()
     mCommandMessageWriter.Reset();
     mCommandMessageBuf = nullptr;
 
-    if (mpExchangeCtx != nullptr)
-    {
-        mpExchangeCtx->Abort();
-        mpExchangeCtx = nullptr;
-    }
+    ClearExistingExchangeContext();
```

Comment:
sure, I will do that. I will start to split it into small prs to accelerate the reason, at the same time, I will keep/update the big PRs so that folks know how the overall e2e works when reviewing the small prs. thanks

Reply:
Thanks!  Again, just splitting out all the cleanup would be a big start; splitting up the functional changes might be both harder and less useful, since as you note it can obscure how the pieces fit together.

Reply:
Done, thanks